SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp

```diff
@@ -155,13 +155,21 @@ bool FSpatialGDKEditorModule::CanExecuteLaunch() const
 
 bool FSpatialGDKEditorModule::CanStartSession(FText& OutErrorMessage) const
 {
-	if (!SpatialGDKEditorInstance->IsSchemaGenerated())
+	FSpatialGDKEditor::ESchemaDatabaseValidationResult SchemaCheck = SpatialGDKEditorInstance->ValidateSchemaDatabase();
+	if (SchemaCheck == FSpatialGDKEditor::NotFound)
 	{
 		OutErrorMessage = LOCTEXT("MissingSchema",
 								  "Attempted to start a local deployment but schema is not generated. You can generate it by clicking on "
 								  "the Schema button in the toolbar.");
 		return false;
 	}
+	else if (SchemaCheck == FSpatialGDKEditor::OldVersion)
+	{
+		OutErrorMessage = LOCTEXT("OldSchema",
+								  "Attempted to start a local deployment but schema is out of date. You can generate it by clicking on "
+								  "the Schema button in the toolbar.");
```

Comment:
After the schema changes related to v15, it wasn't enough to just regenerate schema, you had to first delete the schema database so the generated schema files would be removed. Can we trigger that automatically when generating schema, maybe specifically when the old version is `BeforeVersionSupportAdded`?

Reply:
Ah interesting - yeah will take a look

Reply:
I've modified the schema generation triggering so that if the schema database is old, it'll trigger a full update. We're subsequently put a fix in so that stale schema files will no longer cause an issue in v15 (https://github.com/spatialos/UnrealGDK/pull/2775), so I believe we're covered all bases now.